apps/admin/backend/src/app.test.ts

```diff
@@ -496,3 +497,170 @@ test('auth before election definition has been configured', async () => {
+  });
+  expect(result.isOk()).toEqual(false);
+  expect(result.isErr()).toEqual(true);
+
+  const err = result.err();
+  assert(err);
+  expect(err.type).toEqual('database');
+  expect(err.message).toEqual(
+    `Failed to save system settings to db. Error: ${errorStringToFailWith}`
+  );
+
+  // Logger call 1 is made by configureMachine when loading the election definition
+  expect(logger.log).toHaveBeenNthCalledWith(
+    2,
+    LogEventId.SystemSettingsSaveInitiated,
+    'system_administrator',
+    { disposition: 'na' }
+  );
+  expect(logger.log).toHaveBeenNthCalledWith(
+    3,
+    LogEventId.SystemSettingsSaved,
+    'system_administrator',
+    { disposition: 'failure', error: errorStringToFailWith }
+  );
+}
+
+test('setSystemSettings returns an error when store.addSystemSettings fails', async () => {
+  const exampleErrorString = 'db error at addSystemSettings';
+  await runStoreFailureTest(exampleErrorString, (workspace: Workspace) => {
+    /* eslint-disable no-param-reassign */
+    workspace.store.addSystemSettings = jest.fn(() => {
+      throw new Error(exampleErrorString);
+    });
+  });
+});
+
+test('setSystemSettings returns an error when store.setCurrentSystemSettingsId fails', async () => {
+  const exampleErrorString = 'db error at setCurrentSystemSettingsId';
+  await runStoreFailureTest(exampleErrorString, (workspace: Workspace) => {
+    /* eslint-disable no-param-reassign */
+    workspace.store.setCurrentSystemSettingsId = jest.fn(() => {
+      throw new Error(exampleErrorString);
+    });
+  });
+});
+
+test('setSystemSettings returns an error for malformed input', async () => {
+  const { apiClient, auth } = buildTestEnvironment();
+
+  const { electionDefinition } = electionMinimalExhaustiveSampleFixtures;
+  await configureMachine(apiClient, auth, electionDefinition);
+
+  mockSystemAdministratorAuth(auth);
+
+  const malformedInput = {
+    invalidField: 'hello',
+  } as const;
+
+  const result = await apiClient.setSystemSettings({
+    systemSettings: JSON.stringify(malformedInput),
+  });
+  expect(result.isOk()).toEqual(false);
+  expect(result.isErr()).toEqual(true);
+
+  const err = result.err();
+  assert(err);
+  expect(err.type).toEqual('parsing');
+  expect(JSON.parse(err.message)).toMatchObject([
+    {
+      code: 'invalid_type',
+      expected: 'boolean',
+      received: 'undefined',
+      path: ['pollWorkerCardPinsEnabled'],
+      message: 'Required',
+    },
+  ]);
+});
+
+test('getCurrentSystemSettings happy path', async () => {
+  const { apiClient, auth } = buildTestEnvironment();
+
+  const { electionDefinition, systemSettings } =
+    electionMinimalExhaustiveSampleFixtures;
+  await configureMachine(apiClient, auth, electionDefinition);
+
+  mockSystemAdministratorAuth(auth);
+
+  // configure with well-formed system settings
+  const setResult = await apiClient.setSystemSettings({
+    systemSettings: systemSettings.asText(),
+  });
+  assert(setResult.isOk());
+  setResult.ok();
```

Comment:
The intent was to assert both `isOk()` and `ok()` work on the `Result` but I realize this is a bad half-state because we don't assert on the result of `ok()`. We only implicitly test that it doesn't throw. I'll remove this since it's mostly (or entirely?) redundant with `isOk()`.

Reply:
Also somewhat messy test practice to test `Result` like this at this higher level

Reply:
https://github.com/votingworks/vxsuite/pull/3147/commits/e0d7d876fde6dcf19d0c5a7d7c0a77e202eda086#diff-4a69b2fa57321af9c7b222c3cff0b954bf33b35e6293291bdee466c7bb262842L642